collate/main/grantcm.md

```diff
@@ -0,0 +1,472 @@
+     */
+    public Group toModelType() throws IllegalValueException {
+        return new Group (groupName);
+    }
+```
+###### /java/seedu/address/ui/CommandBox.java
+``` java
+    /**
+     * Handles the Tab button pressed event.
+     */
+    private void handleAutoComplete() {
+        String input = commandTextField.getText();
+        try {
+            String command = commandTrie.attemptAutoComplete(input);
+
+            if (input.equals(command)) {
+                //No command exists in trie
+                setStyleToIndicateCommandFailure();
+                logger.info("Autocomplete failed with input: " + input);
+            } else if (commandSet.contains(command)) {
+                //Able to autocomplete to a correct command
+                this.replaceText(command);
+                logger.info("Autocomplete successful with input: " + input + " to " + command);
+            } else if (commandSet.contains(input)) {
+                //Add parameters
+                this.replaceText(input + command);
+                logger.info("Autocomplete successful with input: " + input + " to " + input + command);
+            }
+        } catch (NullPointerException e) {
+            setStyleToIndicateCommandFailure();
+            logger.info("Autocomplete failed with input: " + input);
+        }
+
+
+    }
+```
+###### /java/seedu/address/ui/GroupCard.java
+``` java
+    public GroupCard (Group group) {
+        super(FXML);
+        this.group = group;
+        name.setText(group.getGroupName());
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof PersonCard)) {
+            return false;
+        }
+
+        // state check
+        GroupCard card = (GroupCard) other;
+        return name.getText().equals(card.name.getText())
+                && group.equals(card.group);
+    }
+```
+###### /java/seedu/address/ui/GroupListPanel.java
+``` java
+    public GroupListPanel(ObservableList<Group> groupList) {
+        super(FXML);
+        setConnections(groupList);
+        registerAsAnEventHandler(this);
+    }
+
+    private void setConnections (ObservableList<Group> groupList) {
+        ObservableList<GroupCard> groupCards = EasyBind.map(
+                groupList, (group) -> new GroupCard(group));
+        groupListView.setItems(groupCards);
+        groupListView.setCellFactory(listView -> new GroupListViewCell());
+        setEventHandlerForSelectionChangeEvent();
+    }
+
+    private void setEventHandlerForSelectionChangeEvent() {
+        groupListView.getSelectionModel().selectedItemProperty()
+                .addListener((observable, oldValue, newValue) -> {
+                    if (newValue != null) {
+                        logger.fine("Selection in person list panel changed to : '" + newValue + "'");
```

Comment:
Good use of logger, however a better use for them is to look for things that can go wrong.

Your logging in this line is fine but take note that using too much logger that are not related to finding bugs may bloat the log file and cause it to be hard to trace.